types/subcall.go

```diff
@@ -1,15 +1,56 @@
 package types
 
-import "encoding/json"
+import (
+	"encoding/json"
+	"fmt"
+)
 
-type replyOn string
+type replyOn int
 
 const (
-	ReplyAlways  replyOn = "always"
-	ReplySuccess replyOn = "success"
-	ReplyError   replyOn = "error"
+	ReplyAlways replyOn = iota
+	ReplySuccess
+	ReplyError
+	ReplyNever
 )
 
+var fromReplyOn = map[replyOn]string{
+	ReplyAlways:  "always",
+	ReplySuccess: "success",
+	ReplyError:   "error",
+	ReplyNever:   "never",
+}
+
+var toReplyOn = map[string]replyOn{
+	"always":  ReplyAlways,
+	"success": ReplySuccess,
+	"error":   ReplyError,
+	"never":   ReplyNever,
+}
+
+func (s replyOn) MarshalJSON() ([]byte, error) {
+	bytes, err := json.Marshal(fromReplyOn[s])
```

Comment:
Let's implement the Stringer interface as Alex suggested: https://golang.org/pkg/fmt/#Stringer

Then this becomes: `json.Marshal(s.String())`